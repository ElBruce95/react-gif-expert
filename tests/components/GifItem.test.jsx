import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {

    const image = {
        title: 'Google',
        url: 'https://www.google.com/',
    };

    test('Debe hacer match con el snapshot', () => {

        const {container} = render( <GifItem { ...image } />);
        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render( <GifItem { ...image } />);
        //screen.debug();

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( image.url );
        expect( alt ).toBe( image.title );

    });

    test('Debe de mostrar el titulo en el componente', () => {

        render( <GifItem { ...image } />);
        expect( screen.getByText( image.title ) ).toBeTruthy();
    });

})