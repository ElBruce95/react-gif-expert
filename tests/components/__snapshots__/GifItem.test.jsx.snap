// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Google"
      src="https://www.google.com/"
    />
    <p>
      Google
    </p>
  </div>
</div>
`;
